lib/cucumber/core/test/case.rb

```diff
@@ -7,14 +7,15 @@ module Cucumber
   module Core
     module Test
       class Case
-        attr_reader :id, :name, :test_steps, :location, :tags, :language, :around_hooks
+        attr_reader :id, :name, :test_steps, :location, :parent_locations, :tags, :language, :around_hooks
 
-        def initialize(id, name, test_steps, location, tags, language, around_hooks = [])
+        def initialize(id, name, test_steps, location, parent_locations, tags, language, around_hooks = [])
```

Comment:
I know this is a huge piece of work. And we are looking into doing some refactoring. So happy to take your input here.

a) don't make any changes (pref setting `parent_locations` to final pos-arg)
b) make parent_locations a kwarg and default it to some value so this is a little safer)
c) begin to pass in a hash of options -> Eventually this is what we probably should be doing for huge objects like this

Reply:
Hmm, let me think.

a) I don't like this, because it seems like `locations` and `parent_locations` are conceptually related, so I would feel icky (technical term) to not have them physically co-located.
b) I don't like this, because the data model is actually **incomplete** without `#parent_locations`. Omitting it in v4.0 is what led to this regression that I've been trying to fix for years!
c) I think this is the "correct" way forward, but I'd suggest they should be kwargs. Most should be required, and some we could default to empty arrays, like `tags`, perhaps.

I'm willing to take on this refactor, but I'm leaning towards doing it in a separate PR that builds on-top of this branch, to keep this PR focused. Should I proceed?

Reply:
Again, happy for things to be deferred. This is a bug fix so I don't mind if we introduce more tech debt here since we're VERY MUCH not rubocop compliant. Obviously down the line we will be